File: app/src/main/java/com/ladsoft/bakingapp/di/module/ServiceBuilder.java
package com.ladsoft.bakingapp.di.module;

import com.ladsoft.bakingapp.service.IngredientUpdateService;
import com.ladsoft.bakingapp.service.di.IngredientUpdateServiceModule;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ServiceBuilder {
    @ContributesAndroidInjector(modules = {IngredientUpdateServiceModule.class})
    abstract IngredientUpdateService providesIngredientUpdateService(); 
}

File: app/src/main/java/com/ladsoft/bakingapp/di/module/ServiceBuilder.kt
package com.ladsoft.bakingapp.di.module

import com.ladsoft.bakingapp.service.IngredientUpdateService
import com.ladsoft.bakingapp.service.di.IngredientUpdateServiceModule
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class ServiceBuilder {
    @ContributesAndroidInjector(modules = [IngredientUpdateServiceModule::class])
    abstract fun providesIngredientUpdateService(): IngredientUpdateService
}